lib/evmone/eof.cpp

```diff
@@ -129,45 +422,114 @@ std::variant<EOF1Header, EOFValidationError> validate_eof1(
         return *error;
 
     const auto& section_headers = std::get<EOFSectionHeaders>(section_headers_or_error);
-    EOF1Header header{section_headers[CODE_SECTION], section_headers[DATA_SECTION]};
+    const auto& code_sizes = section_headers[CODE_SECTION];
+    const auto data_size =
+        section_headers[DATA_SECTION].empty() ? uint16_t{0} : section_headers[DATA_SECTION][0];
```

Comment:
Data section is mandatory
```suggestion
        section_headers[DATA_SECTION][0];
```